client/boot/common.js

```diff
@@ -29,10 +29,9 @@ import authController from 'auth/controller';
 const debug = debugFactory( 'calypso' );
 
 const switchUserLocale = ( currentUser, reduxStore ) => {
-	const localeSlug = currentUser.get().localeSlug;
-
-	if ( localeSlug ) {
-		reduxStore.dispatch( setLocale( localeSlug ) );
+	const currentUserData = currentUser.get();
```

Comment:
We can destructure it here:
```
const { localeSlug, localeVariant } = currentUser.get();
```
so that we don't have to re-access the same fields below.